packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts

```diff
@@ -0,0 +1,366 @@
+   */
+  IN = 'In',
+
+  /**
+   * Checks if the `match` is not included in the target array of strings
+   */
+  NOT_IN = 'NotIn',
+
+  /**
+   * Checks if the `match` starts with one of the target array of strings
+   */
+  STARTS_WITH = 'StartsWith',
+
+  /**
+   * Checks if the `match` is greater than the target number
+   */
+  GREATER_THAN = 'GreaterThan',
+
+  /**
+   * Checks if the `match` is less than the target number
+   */
+  LESS_THAN = 'LessThan',
+
+  /**
+   * Checks if the `match` is equal to the target number
+   */
+  EQUAL_TO = 'EqualTo',
+
+  /**
+   * Checks if the `match` is not equal to the target number
+   */
+  NOT_EQUAL_TO = 'NotEqualTo',
+
+  /**
+   * Checks if the `match` field is present in the log event
+   */
+  IS_PRESENT = 'IsPresent',
+}
+
+/**
+ * A Filter to scope down the log events included in an Insight Rule
+ */
+export class Filter {
+
+  /**
+   * Convenience function to generate an IN Filter
+   */
+  public static in(match: string, target: string[]): Filter {
+    return new Filter(match, FilterOperator.IN, target);
+  }
+
+  /**
+   * Convenience function to generate a NOT_IN Filter
+   */
+  public static notIn(match: string, target: string[]): Filter {
+    return new Filter(match, FilterOperator.NOT_IN, target);
+  }
+
+  /**
+   * Convenience function to generate a STARTS_WITH Filter
+   */
+  public static startsWith(match: string, target: string[]): Filter {
+    return new Filter(match, FilterOperator.STARTS_WITH, target);
+  }
+
+  /**
+   * Convenience function to generate a GREATER_THAN Filter
+   */
+  public static greaterThan(match: string, target: number): Filter {
+    return new Filter(match, FilterOperator.GREATER_THAN, target);
+  }
+
+  /**
+   * Convenience function to generate a LESS_THAN Filter
+   */
+  public static lessThan(match: string, target: number): Filter {
+    return new Filter(match, FilterOperator.LESS_THAN, target);
+  }
+
+  /**
+   * Convenience function to generate an EQUAL_TO Filter
+   */
+  public static equalTo(match: string, target: number): Filter {
+    return new Filter(match, FilterOperator.EQUAL_TO, target);
+  }
+
+  /**
+   * Convenience function to generate a NOT_EQUAL_TO Filter
+   */
+  public static notEqualTo(match: string, target: number): Filter {
+    return new Filter(match, FilterOperator.NOT_EQUAL_TO, target);
+  }
+
+  /**
+   * Convenience function to generate an IS_PRESENT Filter
+   */
+  public static isPresent(match: string, target: boolean): Filter {
+    return new Filter(match, FilterOperator.IS_PRESENT, target);
+  }
+
+  /**
+   * The log event field to assess in the filter
+   */
+  readonly match: string;
+
+  /**
+   * The type of Filter to apply
+   */
+  readonly operator: FilterOperator;
+
+  /**
+   * The target field to compare the `match` against
+   */
+  readonly target: string[] | number | boolean;
+
+  private constructor(match: string, operator: FilterOperator, target: string[] | number | boolean) {
```

Comment:
I don't like having a union type here, and in fact it shouldn't be jsii-compatible. Not sure how you're getting a passing build now that I'm thinking of it. I think it needs to be another enum-like class:

```ts
export class Target {
  public static STRING_LIST(list: string[]) {}
  public static NUMBER(num: number) {}
  public static BOOL(bool: boolean) {}
  private constructor(target: any) {}
  public render(): any {
    return target;
  }
}
```